iosCalorieTrackrSprintChallenge/iosCalorieTrackrSprintChallenge/ViewControllers/CalorieTrackRViewController.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  CalorieTrackRViewController.swift
+//  iosCalorieTrackrSprintChallenge
+//
+//  Created by BrysonSaclausa on 10/10/20.
+//
+
+import UIKit
+import CoreData
+import SwiftChart
+
+
+class CalorieTrackRViewController: UIViewController {
+    
+    // MARK: - IBoutlets
+    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet private weak var chartView: Chart!
+    
+    // MARK: - Properties
+    
+    var calorieIntakeArray: [CalorieIntake] {
+        let fetchRequest: NSFetchRequest<CalorieIntake> = CalorieIntake.fetchRequest()
+        let context = CoreDataStack.shared.mainContext
+        do {
+            return try context.fetch(fetchRequest)
+        } catch {
+            print("error")
+            return []
+        }
+    }
+    
+    // MARK: - Chart Implementation
+    var caloriesAdded: CalorieIntake?
+    var calories: [CalorieIntake] = []
+    var chartNumbers: [Double] = []
+    
+    func setupChart() {
+        chartView.axesColor = .systemTeal
+        chartView.gridColor = .systemRed
+        chartView.backgroundColor = .systemBackground
+    }
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.reloadData()
+        setupChart()
+        createObserver()
+        
+    }
+    
+    // MARK: - Observer
+    func createObserver() {
+        NotificationCenter.default.addObserver(self, selector: #selector(refreshViews(notification:)), name: .doneWasTapped, object: nil)
+        print("OBSERVING")
+    }
+   
+    @objc func refreshViews(notification: Notification) {
+        tableView.reloadData()
+    }
```

Comment:
You can add some logic here to loop through your calorieIntakeArray and get the calories and add them to the chart. Be sure set the chart views series to an empty array so it doesn't double up your entries.